tensorflow/python/compiler/tensorrt/test/trt_engine_op_shape_test.py

```diff
@@ -0,0 +1,94 @@
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+"""This script test input and output shapes and dtype of the TRTEngineOp."""
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+from tensorflow.python.compiler.tensorrt.test import tf_trt_integration_test_base as trt_test
+from tensorflow.python.framework import dtypes
+from tensorflow.python.framework import tensor_shape
+from tensorflow.python.platform import test
+
+from tensorflow.python import saved_model
+from tensorflow.python.saved_model import signature_constants
+from tensorflow.python.saved_model import tag_constants
+
+from tensorflow.python.ops import array_ops
+from tensorflow.python.ops import math_ops
+from tensorflow.python.ops import nn
+
+
+class TRTEngineOpInputOutputShapeTest(trt_test.TfTrtIntegrationTestBase):
+  """Testing conversion of Conv2D (data_format=NCHW) in TF-TRT conversion."""
+
+  def GraphFn(self, inp):
+    b = array_ops.squeeze(inp, axis=[2])
+    c = nn.relu(b)
+    d1 = c + c
+    d2 = math_ops.reduce_sum(d1)
+
+    d1 = array_ops.identity(d1, name="output_0")
+    d2 = array_ops.identity(d2, name="output_1")
+    return d1, d2
+
+  def GetParams(self):
+    # TODO(aaroey): test graph with different dtypes.
+    return self.BuildParams(self.GraphFn, dtypes.float32, [[1, 2, 1, 4]],
+                            [[1, 2, 4], []])
+
+  def _GetInferGraph(self, *args, **kwargs):
+    trt_saved_model_dir = super(
+        TRTEngineOpInputOutputShapeTest, self
+    )._GetInferGraph(*args, **kwargs)
+
+    def get_func_from_saved_model(saved_model_dir):
+      saved_model_loaded = saved_model.load.load(
+          saved_model_dir, tags=[tag_constants.SERVING])
+      graph_func = saved_model_loaded.signatures[
+          signature_constants.DEFAULT_SERVING_SIGNATURE_DEF_KEY]
+      return graph_func, saved_model_loaded
+
+    func, loaded_model = get_func_from_saved_model(trt_saved_model_dir)
+
+    input_shape = func.inputs[0].shape
+    if isinstance(input_shape, tensor_shape.TensorShape):
+      input_shape = input_shape.as_list()
+
+    output_shapes = [
+      out_shape.shape.as_list()
+      if isinstance(out_shape.shape, tensor_shape.TensorShape) else
+      out_shape.shape
+      for out_shape in func.outputs
+    ]
+
+    assert (func.inputs[0].dtype == dtypes.float32)
+    assert (func.outputs[0].dtype == dtypes.float32)
+    assert (func.outputs[1].dtype == dtypes.float32)
+
+    assert (input_shape == [None, 2, 1, 4])
+    assert (output_shapes[0] == [None, 2, 4])
+    assert (output_shapes[1] == [])
```

Comment:
We don't use assert to check test result, instead use self.assertEqual etc.

Reply:
Fixed